app/uk/gov/hmrc/cgtpropertydisposals/service/SubscriptionService.scala

```diff
@@ -53,4 +67,86 @@ class SubscriptionServiceImpl @Inject()(connector: SubscriptionConnector)(implic
       }
     }
 
+  override def getSubscription(cgtReference: CgtReference)(
+    implicit hc: HeaderCarrier
+  ): EitherT[Future, Error, SubscriptionDisplayResponse] =
+    connector.getSubscription(cgtReference).subflatMap { response =>
+      lazy val identifiers =
+        List(
+          "id"                -> cgtReference.value,
+          "DES CorrelationId" -> response.header("CorrelationId").getOrElse("-")
+        )
+
+      if (response.status === 200)
+        response
+          .parseJSON[DesSubscriptionDisplayDetails]()
+          .flatMap(toSubscriptionDisplayRecord(_, cgtReference))
+          .leftMap(Error(_, identifiers: _*))
+      else {
+        Left(Error(s"call to subscription display api came back with status ${response.status}"))
+      }
+    }
+
+  def toSubscriptionDisplayRecord(
+    desSubscriptionDisplayDetails: DesSubscriptionDisplayDetails,
+    cgtReference: CgtReference
+  ): Either[String, SubscriptionDisplayResponse] = {
+
+    val desNonIsoCountryCodes: List[CountryCode] =
+      config.underlying.get[List[CountryCode]]("des.non-iso-country-codes").value
+
+    val addressValidation: Validation[Address] = AddressDetails.fromDesAddressDetails(
+      desSubscriptionDisplayDetails.subscriptionDetails.addressDetails
+    )(desNonIsoCountryCodes)
+
+    val nameValidation: Validation[Either[TrustName, IndividualName]] = Name.nameValidation(
+      desSubscriptionDisplayDetails.subscriptionDetails.individual,
+      desSubscriptionDisplayDetails.subscriptionDetails.trustee
+    )
+
+    (addressValidation, nameValidation)
+      .mapN {
+        case (address, name) =>
+          SubscriptionDisplayResponse(
+            name,
+            desSubscriptionDisplayDetails.subscriptionDetails.contactDetails.emailAddress.flatMap(e => Some(Email(e))),
+            address,
+            ContactName(desSubscriptionDisplayDetails.subscriptionDetails.contactDetails.contactName),
+            cgtReference,
+            desSubscriptionDisplayDetails.subscriptionDetails.contactDetails.phoneNumber
+              .flatMap(t => Some(TelephoneNumber(t))),
```

Comment:
couldn't you just do `.map(t => Some(TelephoneNumber(t)))` instead of a `flatMap` here? Same for the email above

Reply:
change it to use map instead - I like flatMap as is tells me the functor I am dealing explicitly in the code